### ISSUE TITLE
Getting FS0193 with flexible types

### ISSUE THREAD

**Issue by gusty** [dotnet/fsharp#1713]
I was playing a bit with flexible types and by accident found this strange behaviour:
#### Repro steps

In a script paste this code

    type A = class end
    type B = class inherit A end
    type C = class inherit B  end

    type D  =
        inherit C
        static member M x = x

    let inline Inv (x : #C, y) = y

execute it and you will get:

    error FS0193: The invoked member is not supported before the type is created.

try removing one hierarchy:

    type B = class end
    type C = class inherit B  end

    type D  =
        inherit C
        static member M x = x

    let inline Inv (x : #C, y) = y

execute it and you will get:

    error FS0193: internal error: Could not load type 'C' from assembly 'FSI-ASSEMBLY, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'.
#### Expected behavior

It was expected to compile

#### Actual behavior

In both cases we get a FS0193 but with different text in the error description.

#### Known workarounds

- Remove another hierarchy level
- Add a constructor to the classes

#### Related information 

* Operating system: Windows 10
* Branch: Master
* .NET Runtime, CoreCLR or Mono Version: .NET 4.6.01586
* Editing Tools (e.g. Visual Studio Version) : VS Professional 2015 14.0.25123.00 Update 2
* Links to F# RFCs or entries on http://fslang.uservoice.com : N/A
* Links to performance testing scripts : N/A
* Indications of severity : Low

**Comment by abelbraaksma** [comment]
FYI: I can reproduce this on Update 3 as well, and indeed, adding the ctors compiles. Some more observations:

* If you compile it in VS2015 and reference the compiled DLL with `#r` in FSI, it loads the type hierarchy just fine, in other words, this error seems to *only pop up in FSI*.
* Removing `inline` still raises FS0193
* Removing the flexible type marker `#` it runs fine in FSI.
* Placing `Inv` under `D` as (static) member compiles just fine
* Placing `Inv` in its own module (but in the same file) compiles just fine
* Helping the compiler with `and` does not help, nor does messing with the order of declaration

Here's another surprising workaround:

Select and *Send to interactive* this part first:

    type A = class end
    type B = class inherit A end
    type C = class inherit B  end

    type D  =
        inherit C
        static member M x = x

Then select and *Send to interactive* the final line:

    let inline Inv (x : #C, y) = y

No errors this time around.

Bottom line: it *only* seems to happen if you send the whole code segment to FSI.

**Comment by gmpl** [comment]
Thanks @abelbraaksma 
Yes, indeed I forgot to remove the <code>inline</code> from the repro.
Since there are many workarounds I don't think there's any priority on fixing this but a bug is a bug and they should be filed in a bug database and ideally investigated.

**Comment by gmpl** [comment]
@abelbraaksma Actually after these workarounds the repro compiles fine but it still fails later.

For instance if you later define:

    type E =
        inherit D  

        static member M (t : D) = ()

It fails, regardless of all the workarounds that doesn't involve modifying the base classes.

**Comment by zanedp** [comment]
I am getting this error and, as best I can tell, I'm not doing anything with flexible types or extending a class without a constructor. 

If I modify my generated `load-project-debug.fsx` so that every line has a `#load` on it, it will work if I execute all the contents of that file at once. However, if I `#load` that file (with the `#load` on each line), it fails with:
`error FS0193: internal error: Could not load type 'FSI_0002.<mymodule>' from assembly 'FSI-ASSEMBLY, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'.`

Is there something I can enable to help dig up more information about what's causing the problem?

Version is `Microsoft (R) F# Interactive version 14.0.23413.0`

**Comment by dsyme** [comment]
@zanedp Could you open a new bug and submit a repro for what you're seeing please?

**Comment by dsyme** [comment]
Fixed by `--multiemit+` which is on by default for .NET Core
